src/common/types.ts

```diff
@@ -82,3 +83,41 @@ export function asIntegerString(raw: unknown): string {
   }
   return clean
 }
+
+export const asWcProps = asObject({
+  uri: asString,
+  language: asMaybe(asString),
+  token: asMaybe(asString)
+})
+
+export type WcProps = ReturnType<typeof asWcProps>
+
+const asWcDappDetails = asObject({
+  peerId: asString,
+  peerMeta: asObject({
+    description: asString,
+    url: asString,
+    icons: asArray(asString),
+    name: asString
+  }),
+  chainId: asOptional(asNumber, 1)
+})
+
+export type WcDappDetails = {
+  timeConnected: number
+} & ReturnType<typeof asWcDappDetails>
+
+export type Dapp = { timeConnected: number } & WcProps & WcDappDetails
```

Comment:
The `WcDappDetails` already include `timeConnected`, so it's redundant to have it here too?